src/main/java/com/empowerops/linqalike/Linq.java

```diff
@@ -209,12 +216,12 @@ boolean containsElement(Iterable<? extends TElement> sourceElements,
     @SafeVarargs
     public static <TElement>
     Queryable<TElement> union(Iterable<? extends TElement> left, TElement... toInclude) {
-        return new UnionQuery<>(left, from(toInclude), performEqualsUsing(identity()));
+        return new UnionQuery<>(left, from(toInclude), CommonDelegates.DefaultEquality);
```

Comment:
this was a little annoying thing that I don't know If i could turn up in testing.